src/main/java/seedu/address/logic/parser/FindCommandParser.java

```diff
@@ -4,21 +4,32 @@
 
 import java.util.Arrays;
 
+import seedu.address.logic.commands.eventCommands.FindEventCommand;
+import seedu.address.logic.commands.personCommands.Command;
 import seedu.address.logic.commands.personCommands.FindCommand;
 import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.types.common.EventNameContainsKeywordsPredicate;
 import seedu.address.model.types.common.NameContainsKeywordsPredicate;
 
 /**
  * Parses input arguments and creates a new FindCommand object
  */
-public class FindCommandParser implements Parser<FindCommand> {
+public class FindCommandParser implements Parser<Command> {
 
     /**
      * Parses the given {@code String} of arguments in the context of the FindCommand
      * and returns a FindCommand object for execution.
      * @throws ParseException if the user input does not conform the expected format
      */
-    public FindCommand parse(ModelType modelType, String args) throws ParseException {
+    public Command parse(ModelType modelType, String args) throws ParseException {
+        if (modelType == ModelType.PERSON) {
+            return parseForPerson(args);
+        } else {
```

Comment:
else shouldnt be a catch all right? ie. should have if statement for modeltype.event. but just a small nitpick